src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
Thanks for this @olevett. Yeah, agreed, we should avoid alloc'ing where possible, especially at such a low level of the framework. Every little bit helps and all. I know we haven't been particularly vigilant about this kind of thing elsewhere, but it might be time to tighten up. This will be easier to manage once we formalise the inclusion of Benchmarkdotnet into the tests.